models/migrations/v163.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"xorm.io/xorm"
+)
+
+type oauth2grant struct {
+	Scope string
+	Nonce string
+}
+
+func (grant *oauth2grant) TableName() string {
+	return "oauth2_grant"
+}
+
+func addScopeAndNonceColumnsToOAuth2Grant(x *xorm.Engine) error {
+
+	if err := x.Sync2(new(oauth2grant)); err != nil {
```

Comment:
I am getting some errors in the mysql integration tests regarding this migration.  Unfortunately I am not well versed in golang, XORM, or this code base well enough to have a simple fix.

Can someone give me a pointer to what is wrong?

Reply:
Should my type here be a "snapshot" of what OAuth2Grant looked like in this version instead?  I would guess that would fix it.  But on the other hand the other "migrations" don't really seem to do this, so I feel a bit lost.

Reply:
I guess the problem is that the table was not added as a previous migration, and I guess the oauth2_grant table will be created on first use only.  So it would fail unless the table already existed.

I changed the code to have a "snapshot" of what the table looked like.  I suppose that should be the more correct way to do it.

Reply:
The build passed with this change, so I'll resolve this for now.

Please let me know if I am completely off track here.

Reply:
I think the first character of struct name should be capitals

Reply:
And  I think 
```
code string `xorm:"TEXT"`
```
should be better because default it's `VARCHAR(255)`

Reply:
@lunny The reason I kept the struct name in lower case was because I felt it unnecessary to export the type since it will only be used inside the migration file, but I don't care strongly either way.  I have updated it now.

I also changed the strings to be `TEXT` instead of `VARCHAR(255)`, as suggested.